docs/features/multi-cluster-replication.md

```diff
@@ -0,0 +1,64 @@
+# Multi-cluster replication
+
+Temporal supports multi-cluster replication. This feature allows you to replicate specific temporal namespaces to a different temporal cluster. This is useful for disaster recovery, or to have a temporal cluster in a different region for latency reasons, or if you want to upgrade the temporal history shard count.
+
+## How it works
+
+To set up multi-cluster replication using the temporal operator, you must first enable global namespaces on the clusters you wish to support, and then assign them a unique failover version.
+This can be configured via the `spec.replicaton` of the `TemporalCluster` resource. Temporal operator automatically configures the remaining fields, and currently hard-codes the failover
+increment to 100. If a cluster fails, the remaining clusters will elect a new primary cluster based with the lowest failover version. The original cluster, if it comes back online, will
+be assigned a new failover version, which is always the lowest multiple of the failover increment (+ initialFailoverVersion) that is greater than the leader cluster's failover version.
+
+```yaml
+apiVersion: temporal.io/v1beta1
+kind: TemporalCluster
+metadata:
+  name: prod
+spec:
+  replication:
+    enableGlobalNamespace: true
+    initialFailoverVersion: 1
+```
+
+## Starting replication
+
+Once the two clusters are configured, simply set up connections between them using the temporal CLI. In the future, there may be a way to do this via the operator.
+
+```bash
+# port forward to the frontend of the primary cluster
+kubectl port-forward primary-frontend 7233:7233
+
+temporal operator cluster upsert --frontend-address secondary-cluster.namespace.svc.cluster.local:7233 --enable-connection true
+
+# port forward to the frontend of the secondary cluster
+kubectl port-forward secondary-frontend 7233:7233
+
+tepmoral operator cluster upsert --frontend-address primary-cluster.namespace.svc.cluster.local:7233 --enable-connection true
+```
+
+## Replicating namespaces
+
+Once the clusters are connected, you can replicate namespaces between them. This can be done via the temporal CLI, or via the operator. Simply create a new namespace resource for just the primary clusterRef, and add the secondary cluster to the list of clusters. The namespace will be added to the secondary cluster automatically, with all the same settings, and start receiving updates.
+
+```yaml
+apiVersion: temporal.io/v1beta1
+kind: TemporalNamespace
+metadata:
+  name: primary-namespace
+spec:
+  clusterRef:
+    name: primary
+  clusters:
+    - primary
+    - secondary
+  activeClusterName: secondary
+  isGlobalNamespace: true
+```
+
+| **🚨 Note**: Replication does not eagerly push records to the other cluster. If you set up replication with existing workflows on the primary node, make sure to query every workflow on the original node so that they are propagated to prevent data loss.
```

Comment:
Wording is a little awkward here. The point is it only replicates workflows as they are interacted with, so enabling replication will not automatically replicate old workflows. For cases like trying to increase the shard count, this is important as you need to make sure each workflow has been evaluated at least once after replication has been set up.

If the wording in the PR isn't clear I can revise.